graql/language/update.feature

```diff
@@ -0,0 +1,178 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+#noinspection CucumberUndefinedStep
+Feature: Graql Delete Query
+
+  Background: Open connection and create a simple extensible schema
+    Given connection has been opened
+    Given connection does not have any database
+    Given connection create database: grakn
+    Given connection open schema session for database: grakn
+    Given session opens transaction of type: write
+
+    Given graql define
+      """
+      define
+      person sub entity,
+        plays friendship:friend,
+        plays parenthood:parent,
+        plays parenthood:child,
+        owns name,
+        owns ref @key;
+      friendship sub relation,
+        relates friend,
+        owns ref @key;
+      parenthood sub relation,
+        relates parent,
+        relates child;
+      name sub attribute, value string;
+      ref sub attribute, value long;
+      """
+    Given transaction commits
+
+    Given connection close all sessions
+    Given connection open data session for database: grakn
+    Given session opens transaction of type: write
+
+
+  ##########
+  # THINGS #
+  ##########
+
+  Scenario: simple update
+    Given get answers of graql insert
+      """
+      insert
+        $x isa person, has name "Alex", has ref 0;
+        $y isa person, has name "Alex", has ref 1;
+      """
+    Given uniquely identify answer concepts
+      | x         | y         |
+      | key:ref:0 | key:ref:1 |
+    Given transaction commits
+    Given session opens transaction of type: write
+    When graql update
+      """
+      match
+        $x isa person, has ref 1, has $n;
+        $n isa name;
+      delete
+        $x has $n;
+      insert
+        $x has name "Bob";
+      """
+    Then transaction commits
+    When session opens transaction of type: read
+    When get answers of graql match
+      """
+      match $x isa person, has name $n;
+      """
+    Then uniquely identify answer concepts
+      | x          | n               |
+      | key:ref:0  | value:name:Alex |
+      | key:ref:1  | value:name:Bob  |
+
+  Scenario: Deleting the last roleplayer of a relation means it cannot be updated
+    Given get answers of graql insert
+      """
+      insert
+      $x isa person, has name "Alex", has ref 0;
+      $y isa person, has name "Bob", has ref 1;
+      $r (friend: $x, friend:$y) isa friendship, has ref 0;
+      """
+    Given transaction commits
+    Given session opens transaction of type: write
+    When graql update; throws exception
+      """
+      match
+        $x isa person, has name "Alex";
+        $y isa person, has name "Bob";
+        $r (friend: $x) isa friendship;
+      delete
+        $r (friend: $x);
+      insert
+        $r (friend: $y);
```

Comment:
Good test case, are there any other invalid states we could put ourselves into? We could take inspiration from our other existing tests.